paddle/fluid/operators/batch_norm_op.cc

```diff
@@ -496,8 +512,26 @@ class BatchNormGradKernel<platform::CPUDeviceContext, T>
     int scale_coefff = use_global_stats ? 1 : N * sample_size;
     const auto scale_inv_var_nhw = scale_arr * inv_var_arr / scale_coefff;
 
+    // inplace calculation
+    // Y:  ((x - est_mean) * (inv_var) * scale + bias
+    //   formula transform ====>
+    //   (x * inv_var * scale) + (bias - est_mean * inv_var * scale)
+    // X: (y - bias) / scale / (inv_var) + est_mean
+    //   formula transform ====>
+    //    (y - bias) / (scale * inv_var) + est_mean
     switch (data_layout) {
       case DataLayout::kNCHW: {
+        if (is_inplace) {
+          auto px = const_cast<Tensor &>(*x);
```

Comment:
Why `const_cast`? If in-place happens, try to use `auto &px = *y`.

Reply:
Done, thanks